src/lib/to_graph.ml

```diff
@@ -1,308 +1,338 @@
-    loop ~pv:prev_node ~nv:next_ref_node alt_lst
+    loop prev next
   in
-  match alt_lst with
-  | (Ms.Start (start_pos, allele) as h) :: t ->
-    let v = seq_elem_to_vertex h in
-    G.add_vertex g v;
-    let bef_start_pos_n_ref, start_pos =
-      split_reference_n_at ~pos:start_pos (S reference)
-        ~e:(fun _pv ->
-              invalid_argf "Couldn't find a start for %s %d, these aren't aligned!"
-                allele start_pos)
-        ~join_split:(fun _ _ -> ())
+  (* How we close back with the reference *)
+  let rec rejoin_after_split ~prev ~next split_pos state ~new_node lst =
+    match split_in ~prev ~next ~visit:do_nothing split_pos with
+    | `AfterLast _            -> solo_loop state new_node lst 
+    | `InGap (_pv, next, ap)  -> ref_gap_loop state new_node next ap lst
+    | `AtNext (_pv, next)     -> add_allele_edge new_node next;
+                                 main_loop state ~prev:new_node ~next lst 
+  (* In the beginning we have not 'Start'ed ->
+     Loop through the Alignment elemends:
+      - discarding Boundary
+      - complaining on anything other than a Start
+      - on a Start find the position in reference and start correct loop *)
+  and start_loop prev_start_ends lst =
+    let rec find_start_loop = function
+      | Boundary _ :: t 
+      | Gap _ :: t      -> find_start_loop t (* Ignore Gaps & Boundaries before Start *)
+      | Start p :: t    -> let vs = G.V.create (S (p, allele)) in
+                           Some (vs, p, t)
+      | []              ->
+        begin
+          match prev_start_ends with
+          | [] -> inv_argf "Failed to find start for %s." allele
+          | ls -> None
+        end
+      | s :: _          -> inv_argf "Encountered %s in %s instead of Start"
+                            (al_el_to_string s) allele
     in
-    (* If we start right before a boundary, we want the start to point at
-       the boundary, so look ahead at a boundary and stop early. *)
-    let start = use_boundary_if_it_came_before ~before:bef_start_pos_n_ref ~after:start_pos in
-    G.add_edge_e g (G.E.create v allele start);
-    insert_alt_sequence v start allele t
-  | _ ->
-    invalid_argf "Non reference sequences doesn't start with a start!"
+    match find_start_loop lst with
+    | None -> prev_start_ends (* fin *)
+    | Some (new_node, start_pos, tl) ->
+        let state = new_node, prev_start_ends in
+        rejoin_after_split ~prev:first_start ~next:first_start start_pos state
+          ~new_node tl
+  (* When the only thing that matters is the previous node. *)
+  and solo_loop ((ps, os) as state) vp = function
+    | []                        -> inv_argf "No End at allele sequence: %s" allele
+    | Start p :: _              -> inv_argf "Another start %d in %s allele sequence." p allele
+    | End pos :: []             -> let ve = G.V.create (E pos) in
+                                   add_allele_edge vp ve;
+                                   (ve,ps) :: os (* fin *)
```

Comment:
I think this should be switched?  `(pe, ve)`? Not sure which one we found.

This is where it might be nice to use a record instead of a tuple, i.e. `{start_vertex; end_vertex}` to make the error harder to make (and force longer names!)